File: Codes/19800/19844.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;

public class Main {
    public static int ans = 0;

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        List<String> list = new ArrayList<>();
        String[] arr = br.readLine().split(" ");
        for (String s : arr) {
            int idx = s.indexOf('-');
            if (idx >= 0) {
                String[] tmp = s.split("-");
                for (String t : tmp)
                    list.add(t);
            } else
                list.add(s);
        }
        for (String s : list)
            process(s);
        System.out.println(ans);
    }

    public static void process(String s) {
        if (s.length() < 3) {
            ans++;
            return;
        }
        char c = s.charAt(0);
        char d = s.charAt(1);
        char e = s.charAt(2);
        if (d == '\'') {
            if ((c == 'c' || c == 'j' || c == 'n' || c == 'm' || c == 't' || c == 's' || c == 'l' || c == 'd')
                    && (e == 'a' || e == 'e' || e == 'i' || e == 'o' || e == 'u' || e == 'h')) {
                ans += 2;
                return;
            }
        } else if (e == '\'') {
            if (c == 'q' && d == 'u' && s.length() >= 4) {
                char f = s.charAt(3);
                if (f == 'a' || f == 'e' || f == 'i' || f == 'o' || f == 'u' || f == 'h') {
                    ans += 2;
                    return;
                }
            }
        }
        ans++;
    }
}

File: Codes/19800/19844.cpp
#include <bits/stdc++.h>
using namespace std;

int ans = 0;

void process(string s);

vector<string> split(string str, char c) {
    vector<string> ret;
    int prev = 0, next = str.find(c, prev);
    while((next = str.find(c, prev)) != string::npos) {
        string s = str.substr(prev, next - prev);
        ret.push_back(s);
        prev = next + 1;
    }
    string s = str.substr(prev);
    ret.push_back(s);
    return ret;
}

int main() {
    ios_base::sync_with_stdio(false); cin.tie(0); cout.tie(0);
    string str;
    getline(cin, str);
    vector<string> list;
    vector<string> arr = split(str, ' ');
    for(string s : arr) {
        int idx = s.find('-');
        if(idx >= 0) {
            vector<string> tmp = split(s, '-');
            for(string t : tmp)
                list.push_back(t);
        } else
            list.push_back(s);
    }
    for(string s : list)
        process(s);
    cout << ans;
    return 0;
}

void process(string s) {
    if(s.size() < 3) {
        ans++;
        return;
    }
    char c = s[0], d = s[1], e = s[2];
    if(d == '\'') {
        if(strchr("cjnmtsld", c) && strchr("aeiouh", e)) {
            ans += 2;
            return;
        }
    } else if(e == '\'') {
        if(c == 'q' && d == 'u' && s.size() >= 4) {
            char f = s[3];
            if(strchr("aeiouh", f)) {
                ans += 2;
                return;
            }
        }
    }
    ans++;
}